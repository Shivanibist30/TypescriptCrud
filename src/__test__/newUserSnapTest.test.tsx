import Enzyme,{shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';



const baseUrl='http://localhost:3000/newusers';

jest.mock('../components/addNewUser/newUser',()=>jest.fn());
const mockFn=jest.fn()
Enzyme.configure({adapter:new Adapter()});
let browser;
// const AssertMatchingImageAndSave = async function () {
//     const image = await page.screenshot();
//     expect(image).toMatchImageSnapshot({
//         failureThreshold: 0,
//         failureThresholdType: 'percent',
//         updatePassedSnapshot: process.env.updateAlways === 'true',
        
//     });
// };

// beforeAll(async()=>{
//     browser = await puppeteer.launch({executablePath:browserPath,
//         headless: false,
//     });
//     const page=await browser.newPage();
//     await page.setViewport({width:960,height:1080});
//     return page;
// })

describe('snapshot newuser',()=>{

    it("snapshot testing",()=>{
        const wrapper=shallow(<button name='addUser' onClick={mockFn}/>);
        wrapper.simulate('click')
        expect(mockFn).toMatchSnapshot()
    });
    

    

});