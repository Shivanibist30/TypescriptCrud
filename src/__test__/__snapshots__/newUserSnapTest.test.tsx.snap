// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot newuser snapshot testing 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
